### ISSUE TITLE
GtfsStorage sometimes not closed properly

### ISSUE THREAD

**Issue by takb** [GIScience/openrouteservice#1325]
Loading a previously prepared public transport profile sometimes produces  
```
14 Mar 09:29:03 ERROR [routing.RoutingProfileManager] - java.util.concurrent.ExecutionException: java.io.IOError: java.io.IOException: Wrong index checksum, store was not closed properly and could be corrupted.
14 Mar 09:29:03 ERROR [routing.RoutingProfileManager] - Failed to initialize RoutingProfileManager instance.
java.util.concurrent.ExecutionException: java.io.IOError: java.io.IOException: Wrong index checksum, store was not closed properly and could be corrupted.
	at java.util.concurrent.FutureTask.report(FutureTask.java:122) ~[?:?]
	at java.util.concurrent.FutureTask.get(FutureTask.java:191) ~[?:?]
	at org.heigit.ors.routing.RoutingProfileManager.initialize(RoutingProfileManager.java:184) ~[classes/:?]
	at org.heigit.ors.routing.RoutingProfileManager.getInstance(RoutingProfileManager.java:75) ~[classes/:?]
	at org.heigit.ors.servlet.listeners.ORSInitContextListener.lambda$contextInitialized$0(ORSInitContextListener.java:42) ~[classes/:?]
	at java.lang.Thread.run(Thread.java:833) ~[?:?]
Caused by: java.io.IOError: java.io.IOException: Wrong index checksum, store was not closed properly and could be corrupted.
	at org.mapdb.StoreDirect.checkHeaders(StoreDirect.java:269) ~[mapdb-1.0.8.jar:?]
	at org.mapdb.StoreDirect.<init>(StoreDirect.java:207) ~[mapdb-1.0.8.jar:?]
	at org.mapdb.DBMaker.extendStoreDirect(DBMaker.java:971) ~[mapdb-1.0.8.jar:?]
	at org.mapdb.DBMaker.makeEngine(DBMaker.java:758) ~[mapdb-1.0.8.jar:?]
	at org.mapdb.DBMaker.make(DBMaker.java:701) ~[mapdb-1.0.8.jar:?]
	at com.graphhopper.gtfs.GtfsStorage.loadExisting(GtfsStorage.java:161) ~[graphhopper-reader-gtfs-v4.1.jar:?]
	at com.graphhopper.gtfs.GraphHopperGtfs.importPublicTransit(GraphHopperGtfs.java:80) ~[graphhopper-reader-gtfs-v4.1.jar:?]
	at com.graphhopper.GraphHopper.postProcessing(GraphHopper.java:1022) ~[graphhopper-core-v4.1.jar:?]
	at com.graphhopper.GraphHopper.load(GraphHopper.java:871) ~[graphhopper-core-v4.1.jar:?]
	at com.graphhopper.GraphHopper.importOrLoad(GraphHopper.java:668) ~[graphhopper-core-v4.1.jar:?]
	at org.heigit.ors.routing.graphhopper.extensions.ORSGraphHopper.importOrLoad(ORSGraphHopper.java:169) ~[classes/:?]
	at org.heigit.ors.routing.RoutingProfile.initGraphHopper(RoutingProfile.java:193) ~[classes/:?]
	at org.heigit.ors.routing.RoutingProfile.<init>(RoutingProfile.java:140) ~[classes/:?]
	at org.heigit.ors.routing.RoutingProfileLoader.call(RoutingProfileLoader.java:39) ~[classes/:?]
	at org.heigit.ors.routing.RoutingProfileLoader.call(RoutingProfileLoader.java:25) ~[classes/:?]
	at java.util.concurrent.FutureTask.run$$$capture(FutureTask.java:264) ~[?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java) ~[?:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539) ~[?:?]
	at java.util.concurrent.FutureTask.run$$$capture(FutureTask.java:264) ~[?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) ~[?:?]
	... 1 more
Caused by: java.io.IOException: Wrong index checksum, store was not closed properly and could be corrupted.
	at org.mapdb.StoreDirect.checkHeaders(StoreDirect.java:269) ~[mapdb-1.0.8.jar:?]
	at org.mapdb.StoreDirect.<init>(StoreDirect.java:207) ~[mapdb-1.0.8.jar:?]
	at org.mapdb.DBMaker.extendStoreDirect(DBMaker.java:971) ~[mapdb-1.0.8.jar:?]
	at org.mapdb.DBMaker.makeEngine(DBMaker.java:758) ~[mapdb-1.0.8.jar:?]
	at org.mapdb.DBMaker.make(DBMaker.java:701) ~[mapdb-1.0.8.jar:?]
	at com.graphhopper.gtfs.GtfsStorage.loadExisting(GtfsStorage.java:161) ~[graphhopper-reader-gtfs-v4.1.jar:?]
	at com.graphhopper.gtfs.GraphHopperGtfs.importPublicTransit(GraphHopperGtfs.java:80) ~[graphhopper-reader-gtfs-v4.1.jar:?]
	at com.graphhopper.GraphHopper.postProcessing(GraphHopper.java:1022) ~[graphhopper-core-v4.1.jar:?]
	at com.graphhopper.GraphHopper.load(GraphHopper.java:871) ~[graphhopper-core-v4.1.jar:?]
	at com.graphhopper.GraphHopper.importOrLoad(GraphHopper.java:668) ~[graphhopper-core-v4.1.jar:?]
	at org.heigit.ors.routing.graphhopper.extensions.ORSGraphHopper.importOrLoad(ORSGraphHopper.java:169) ~[classes/:?]
	at org.heigit.ors.routing.RoutingProfile.initGraphHopper(RoutingProfile.java:193) ~[classes/:?]
	at org.heigit.ors.routing.RoutingProfile.<init>(RoutingProfile.java:140) ~[classes/:?]
	at org.heigit.ors.routing.RoutingProfileLoader.call(RoutingProfileLoader.java:39) ~[classes/:?]
	at org.heigit.ors.routing.RoutingProfileLoader.call(RoutingProfileLoader.java:25) ~[classes/:?]
	at java.util.concurrent.FutureTask.run$$$capture(FutureTask.java:264) ~[?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java) ~[?:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539) ~[?:?]
	at java.util.concurrent.FutureTask.run$$$capture(FutureTask.java:264) ~[?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) ~[?:?]
	... 1 more
``` 

Maybe there is some problem with closing the storage properly, investigate if we have time

**Comment by MichaelsJP** [comment]
closes #1209 ?

**Comment by benmoon1** [comment]
I'm having this issue too in version 8.0. I've tried the test GTFS data from this repo and also files from https://gtfs.pro/en/uk/Department-for-Transport-UK/west-midlands-dft as examples.
